public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -120,23 +119,27 @@ export function runSplitGroupedQueries(datasource: LokiDatasource, requests: Lok
       return;
     }
 
-    subquerySubsciption = datasource
-      .runQuery({ ...requests[requestGroup].request, range, requestId, targets })
-      .subscribe({
-        next: (partialResponse) => {
-          mergedResponse = combineResponses(mergedResponse, partialResponse);
-          if ((mergedResponse.errors ?? []).length > 0 || mergedResponse.error != null) {
-            shouldStop = true;
-          }
-        },
-        complete: () => {
-          subscriber.next(mergedResponse);
-          nextRequest();
-        },
-        error: (error) => {
-          subscriber.error(error);
-        },
-      });
+    const subRequest = { ...requests[requestGroup].request, range, targets };
+    // request may not have a request id
+    if (group.request.requestId) {
+      subRequest.requestId = `${group.request.requestId}_${requestN}`;
```

Comment:
I'm not really sure of the purpose of the `requestId`, but I wonder if we shouldn't always define one, like

```javascript
      subRequest.requestId = subRequest.requestId ? `${group.request.requestId}_${requestN}` : `${requestN}`;
```

Reply:
That would still end up assigning identical request IDs , '1',  to unrelated requests and causing the problem.
Instead we could generate a random `requestId` [here](https://github.com/grafana/grafana/pull/66161/files/3b654bdd3ba5c75aa21d1fffec7f45ca4af577af#diff-623c3cc7d36f876a6b5994d057bc91dbad65060046cddfe6fc68df9de9ca8837R181) if one does not exist. 
But I think if one was not assigned to begin with, it's not really needed at all

Reply:
You're right.

> But I think if one was not assigned to begin with, it's not needed at all

Yes, perfect. I just wanted to double check if we should always have one or not. Thanks!